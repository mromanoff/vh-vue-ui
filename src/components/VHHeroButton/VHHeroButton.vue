<template>
  <button
    :disabled="disabled"
    class="VHHeroButton"
    v-on="$listeners"
  >
    <slot />
  </button>
</template>

<script>
export default {
  name: 'VHHeroButton',

  props: {
    disabled: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style scoped>
.VHHeroButton {
  display: block;
  height: var(--VHHeroButton-height);
  padding: 0 var(--space--xLarge);
  user-select: none;
  text-decoration: none;
  text-transform: uppercase;
  line-height: 1;
  font-size: var(--font-size--medium);
  white-space: nowrap;
  border-radius: calc(var(--VHHeroButton-height) / 2);
  border: var(--VHHeroButton-borderWidth) solid var(--VHHeroButton-borderColor);
  background-color: var(--VHHeroButton-backgroundColor);
  outline: none;
  cursor: pointer;
  color: var(--VHHeroButton-color);
  transition: background-color 100ms ease-out 100ms;
  text-align: center;
}

.VHHeroButton:focus {
  outline: 0;
}

.VHHeroButton:hover,
.VHHeroButton:focus,
.VHHeroButton:active {
  background-color: var(--VHHeroButton-backgroundColor--hover);
}

.VHHeroButton[disabled] {
  opacity: 0.75;
  cursor: not-allowed;
}
</style>
