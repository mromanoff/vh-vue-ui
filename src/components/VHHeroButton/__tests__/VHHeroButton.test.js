import { shallowMount } from '@vue/test-utils';
import VHHeroButton from '../VHHeroButton.vue';

describe('VHHeroButton.vue', () => {
  it('renders default button', () => {
    const wrapper = shallowMount(VHHeroButton);
    expect(wrapper.findAll('.VHHeroButton')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});

it('renders disabled button', () => {
  const wrapper = shallowMount(VHHeroButton, {
    propsData: {
      disabled: true,
    },
  });
  expect(wrapper.findAll('[disabled]')).toHaveLength(1);
  expect(wrapper).toMatchSnapshot();
});
