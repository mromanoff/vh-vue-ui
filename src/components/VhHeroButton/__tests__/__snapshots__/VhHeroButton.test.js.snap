// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhHeroButton.vue renders default button 1`] = `<button class="VhHeroButton"></button>`;

exports[`renders disabled button 1`] = `<button disabled="disabled" class="VhHeroButton"></button>`;
